### PR TITLE
OCT-1861: Cover /recalculate with API Tests

### PR DESCRIPTION
**Pull request by kgarbacinski**
## Definition of Done

1. [ ] Acceptance criteria are met.
2. [ ] PR is manually tested before the merge by developer(s).
    - [ ] Happy path is manually checked.
3. [ ] PR is manually tested by QA when their assistance is required (1).
    - [ ] Octant Areas & Test Cases are checked for impact and updated if required (2).
4. [ ] Unit tests are added unless there is a reason to omit them.
5. [ ] Automated tests are added when required.
6. [ ] The code is merged.
7. [ ] Tech documentation is added / updated, reviewed and approved (including mandatory approval by a code owner, should such exist for changed files).
    - [ ] BE: Swagger documentation is updated.
8. [ ] When required by QA:
    - [ ] Deployed to the relevant environment.
    - [ ] Passed system tests.

---

(1) Developer(s) in coordination with QA decide whether it's required. For small tickets introducing small changes QA assistance is most probably not required.

(2) [Octant Areas & Test Cases](https://docs.google.com/spreadsheets/d/1cRe6dxuKJV3a4ZskAwWEPvrFkQm6rEfyUCYwLTYw_Cc).

### CODE REVIEW

File: backend/tests/api-e2e/test_api_delegation.py [hunk 1]
```diff
@@ -59,5 +59,49 @@ def test_delegation(client: Client, payload: ScoreDelegationPayload):
...
 
 @pytest.mark.api
-def test_check_in_delegation():
-    pass
+def test_recalculate_in_delegation(client: Client, payload: ScoreDelegationPayload):
+    """
+    Recalculation can actually return two different results:
+    - if the delegation does not exist, it will return 400
+    - if the delegation exists, i.e. secondary address exists in the database, it will return 400
+    it's due to the fact that the recalculation is already stoned for a secondary address in our implementation
```
Comment (paulperegud) [inline backend/tests/api-e2e/test_api_delegation.py:67]: > score is already set in stone

Comment (paulperegud) [reply backend/tests/api-e2e/test_api_delegation.py:67]: Or possibly - petrified.

**Review by paulperegud — COMMENTED** [review]
While you are at it, please add one more check to `test_delegate`. After a successful delegation primary and secondary are "burned" for delegation - i.e. an attempt to delegate from USER1 to USER3 should fail.

File: backend/tests/api-e2e/test_api_delegation.py (hunk 1 above)
Comment (kgarbacinski) [reply backend/tests/api-e2e/test_api_delegation.py:78]: added an attempt of doing a redelegation with the same secondary address and checking if it returns a correct response

**Review by paulperegud — APPROVED** [review]

**Review by MateuszStoleckiGLM — APPROVED** [review]

**Pull request merged**